core/tests/unit/ldml/keyboards/meson.build

```diff
@@ -44,15 +59,34 @@ foreach kbd : tests
 
   configure_file(
     command: kmc_cmd + ['build', '@INPUT@', '--out-file', '@OUTPUT@'],
+    input: kbd + '.xml',
     output: kbd + '.kmx',
-    input: kbd + '.xml'
   )
 endforeach
 
 foreach kbd : tests_with_testdata
   configure_file(
     command: kmc_cmd + ['build-test-data', '@INPUT@', '--out-file', '@OUTPUT@'],
+    input: kbd + '-test.xml',
+    output: kbd + '-test.json',
+  )
+endforeach
+
+
+foreach kbd : tests_from_cldr
+  configure_file(
+    command: copy_cmd + ['@INPUT@', '@OUTPUT@'],
+    input: join_paths(ldml_data, kbd + '.xml'),
+    output: kbd + '.xml'
+  )
+  configure_file(
+    command: kmc_cmd + ['build', '@INPUT@', '--out-file', '@OUTPUT@'],
+    input: join_paths(ldml_data, kbd + '.xml'),
+    output: kbd + '.kmx',
+  )
+  configure_file(
+    command: kmc_cmd + ['build-test-data', '@INPUT@', '--out-file', '@OUTPUT@'],
+    input: join_paths(ldml_testdata, kbd + '-test.xml'),
     output: kbd + '-test.json',
-    input: kbd + '-test.xml'
   )
 endforeach
```

Comment:
copypasta and probably more complex than it should be. There are 3 different kinds of sample keyboard files at present. 

The good thing here is it's loading a subset directly out of CLDR!